src/formats/notion-api.ts

```diff
@@ -1934,57 +1981,79 @@ export class NotionAPIImporter extends FormatImporter {
 	}
 
 	/**
-	 * Check if a file should be skipped during import
-	 * This applies to BOTH incremental and full import modes
-	 * 
+	 * Check if a file should be skipped, updated, or created during import
+	 * Supports delta sync by comparing timestamps when incremental import is enabled
+	 *
 	 * @param filePath - Path to the file to check
 	 * @param notionId - Notion ID of the page being imported
+	 * @param pageLastEdited - Last edited timestamp from Notion API
 	 * @param ctx - Import context for reporting
-	 * @returns true if file should be skipped, false otherwise
+	 * @returns Object indicating action: 'skip', 'update', or 'create'
 	 */
-	private async shouldSkipExistingFile(
+	private async shouldSkipOrUpdateFile(
 		filePath: string,
 		notionId: string,
+		pageLastEdited: string,
 		ctx: ImportContext
-	): Promise<boolean> {
+	): Promise<{ action: 'skip' | 'update' | 'create' }> {
 		// Check if file exists
 		const file = this.vault.getAbstractFileByPath(normalizePath(filePath));
 		if (!file || !(file instanceof TFile)) {
-			return false; // File doesn't exist, don't skip
+			return { action: 'create' }; // File doesn't exist
 		}
 
 		// Read file and extract notion-id from frontmatter
 		try {
 			const content = await this.vault.read(file);
 			const notionIdMatch = content.match(/^notion-id:\s*(.+)$/m);
-		
-			if (notionIdMatch) {
-				const existingNotionId = notionIdMatch[1].trim();
-				if (existingNotionId === notionId) {
-					// Same notion-id, skip this file
-					const { basename } = parseFilePath(filePath);
-					ctx.reportSkipped(basename, 'already exists with same notion-id');
-				
-					// IMPORTANT: Register this skipped file in notionIdToPath mapping
-					// This ensures that relation/mention links can find this page even though it wasn't imported in this session
-					// Without this, we would fail to resolve relations to previously imported pages
-					const filePathWithoutExtension = filePath.replace(/\.md$/, '');
-					this.notionIdToPath.set(notionId, filePathWithoutExtension);
-				
-					// IMPORTANT: Scan for unresolved placeholders from previous imports
-					// If the file contains placeholders (relation UUIDs, mentions, synced children) that weren't replaced,
-					// we need to re-collect them so they can be resolved in this import session
-					await this.collectUnresolvedPlaceholders(content, notionId, filePath);
-				
-					return true;
-				}
+
+			if (!notionIdMatch) {
+				return { action: 'create' }; // Different file, will use unique path
+			}
+
+			const existingNotionId = notionIdMatch[1].trim();
+			if (existingNotionId !== notionId) {
+				return { action: 'create' }; // Different notion-id, will rename
+			}
+
+			// Same notion-id found - check if update needed
+			const timestamps = extractNotionTimestamps(content);
+
+			if (!timestamps) {
```

Comment:
I would be nice to combine the two skip cases to help prevent accidents in the future causing their logic to drift.

```ts
if (!timestamps || pageLastEdited > timestamps.updated) {
    // Skip
}
else {
    // Needs update
}
```